torch/distributions/lowrank_multivariate_normal.py

```diff
@@ -0,0 +1,180 @@
+
+        scale_batch_shape = _get_batch_shape(cov_factor, cov_diag)
+        try:
+            batch_shape = torch._C._infer_size(loc.shape[:-1], scale_batch_shape)
+        except RuntimeError:
+            raise ValueError("Incompatible batch shapes: loc {}, cov_factor {}, cov_diag {}"
+                             .format(loc.shape, cov_factor.shape, cov_diag.shape))
+
+        loc_shape = batch_shape + event_shape
+        self.loc = loc.expand(loc_shape)
+        self.cov_factor = cov_factor.expand(loc_shape + cov_factor.shape[-1:])
+        self.cov_diag = cov_diag.expand(loc_shape)
+        self._capacitance_tril = _batch_capacitance_tril(self.cov_factor, self.cov_diag)
+        super(LowRankMultivariateNormal, self).__init__(batch_shape, event_shape,
+                                                        validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return self.loc
+
+    @property
+    def variance(self):
+        return self.cov_factor.pow(2).sum(-1) + self.cov_diag
+
+    @lazy_property
+    def scale_tril(self):
+        # The following identity is used to increase the numerically computation stability
+        # for Cholesky decomposition (see http://www.gaussianprocess.org/gpml/, Section 3.4.3):
+        #     W @ W.T + D = D1/2 @ (I + D-1/2 @ W @ W.T @ D-1/2) @ D1/2
+        # The matrix "I + D-1/2 @ W @ W.T @ D-1/2" has eigenvalues bounded from below by 1,
+        # hence it is well-conditioned and safe to take Cholesky decomposition.
+        n = self._event_shape[0]
+        cov_diag_sqrt_unsqueeze = self.cov_diag.sqrt().unsqueeze(-1)
+        Dinvsqrt_W = self.cov_factor / cov_diag_sqrt_unsqueeze
+        K = torch.matmul(Dinvsqrt_W, Dinvsqrt_W.transpose(-1, -2)).contiguous()
+        K.view(-1, n * n)[:, ::n + 1] += 1  # add identity matrix to K
+        return cov_diag_sqrt_unsqueeze * _batch_potrf_lower(K)
+
+    @lazy_property
+    def covariance_matrix(self):
+        return (torch.matmul(self.cov_factor, self.cov_factor.transpose(-1, -2)) +
+                _batch_vector_diag(self.cov_diag))
+
+    @lazy_property
+    def precision_matrix(self):
+        # We use "Woodbury matrix identity" to take advantage of low rank form::
+        #     inv(W @ W.T + D) = inv(D) - inv(D) @ W @ inv(C) @ W.T @ inv(D)
+        # where :math:`C` is the capacitance matrix.
+        Wt_Dinv = self.cov_factor.transpose(-1, -2) / self.cov_diag.unsqueeze(-2)
+        A = _batch_trtrs_lower(Wt_Dinv, self._capacitance_tril)
+        return (_batch_vector_diag(self.cov_diag.reciprocal()) -
+                torch.matmul(A.transpose(-1, -2), A))
+
+    def rsample(self, sample_shape=torch.Size()):
+        shape = self._extended_shape(sample_shape)
+        eps_W = self.loc.new_empty(shape[:-1] + (self.cov_factor.size(-1),)).normal_()
+        eps_D = self.loc.new_empty(shape).normal_()
+        return self.loc + _batch_mv(self.cov_factor, eps_W) + self.cov_diag * eps_D
```

Comment:
Shouldn't the second term be `self.cov_diag.sqrt() * eps_D`? Can you please add a test that would have caught this?